app/controllers/api/v1/packages_controller.rb

```diff
@@ -161,16 +160,14 @@ def designate_stockit_item(order_id)
       end
 
       def undesignate_partial_item
-        OrdersPackage.undesignate_partially_designated_item(params[:package])
-        @package.undesignate_from_stockit_order
+        designator = Designator.new(@package, params[:package])
+        designator.undesignate
         send_stock_item_response
       end
 
       def designate_partial_item
-        result = OrdersPackage.add_partially_designated_item(
-          order_id: params[:package][:order_id],
-          package_id: params[:package][:package_id],
-          quantity: params[:package][:quantity])
+        designator = Designator.new(@package, params[:package])
+        result = designator.designate
```

Comment:
same here, this can be written as `result = Designator.new(@package, params[:package]).designate`

Reply:
My personal preference in those scenarios is usually to have it as a Service, and not needing an instance at all
e.g `Designator.designate(@package)`

not a big deal though, just worth sharing 😄 
the important part is that the logic was extracted to a single place, the rest is just nitpicking